sklearn/datasets/svmlight_format.py

```diff
@@ -341,8 +341,9 @@ def dump_svmlight_file(X, y, f,  zero_based=True, comment=None, query_id=None,
         Training vectors, where n_samples is the number of samples and
         n_features is the number of features.
 
-    y : array-like, shape = [n_samples]
-        Target values.
+    y : array-like, shape = [n_samples] or [n_samples, n_labels]
+        Target values. Class labels must be an integer or float, or array-like 
+        objects of integer or float for multilabel classifications.
```

Comment:
hi Andreas, please correct me if I'm wrong, but it looks to me that floats are supported for multi-label. Actually in sklearn/datasets/_svmlight_format.pyx, the _load_svmlight_file() method explicitly convert the label y to float (lines 60-72, also below). so at least you can imagine a case where you load a svmlight file using this function, and dump this back to another file. 
I have also tried generating a dataset using make_multilabel_classification(return_indicator = True), and converted the multi-label matrix (variable 'y')to float using y.astype('float'), and plugged it into the dump_svmlight_file() method with "multilabel=True" option, and it works in generating a svmlight file.

thanks and please let me know if you have some other comments.
      
lines from make_multilabel_classification:
  if multilabel:
            if COLON in target:
                target, features = [], line_parts[0:]
            else:
                target = [float(y) for y in target.split(COMMA)]
            target.sort()
            labels.append(tuple(target))